dace/transformation/interstate/loop_overwrite_elimination.py

```diff
@@ -0,0 +1,176 @@
+                    # If pointers are involved or it's not an overwrite, give up
+                    if e.data.dynamic or e.data.wcr is not None:
+                        return False
+                    read_subsets.add(e.data.get_src_subset(e, state))
+                for e in state.in_edges(dn):
+                    # If pointers are involved or it's not an overwrite, give up
+                    if e.data.dynamic or e.data.wcr is not None:
+                        return False
+                    write_subsets.add(e.data.get_dst_subset(e, state))
+                if any(not intersects(rs, ws) for rs in read_subsets for ws in write_subsets):
+                    return False
+
+        # Every write needs to be independent of the loop index.
+        write_subsets = {}
+        for state in self.loop.all_states():
+            for dn in state.data_nodes():
+                if dn.data in unique_set:
+                    continue
+                for e in state.in_edges(dn):
+                    # If pointers are involved or it's not an overwrite, give up
+                    if e.data.dynamic or e.data.wcr is not None:
+                        return False
+
+                    dst_subset = e.data.get_dst_subset(e, state)
+                    for rb, re, _ in dst_subset.ndrange():
+                        str_set = set([str(s) for s in rb.free_symbols.union(re.free_symbols)])
+                        if itervar_dep_syms.intersection(str_set):
+                            return False
+
+                    if dn.data not in write_subsets:
+                        write_subsets[dn.data] = set()
+                    write_subsets[dn.data].add(dst_subset)
+
+        # If an data container is written and read, the last read cannot be the same index as the write, because there is a loop-carried dependency then.
+        for state in self.loop.all_states():
+            for dn in state.data_nodes():
+                if dn.data in unique_set or dn.data not in write_subsets:
+                    continue
+                for e in state.out_edges(dn):
+                    # If pointers are involved or it's not an overwrite, give up
+                    if e.data.dynamic or e.data.wcr is not None:
+                        return False
+
+                    src_subset = copy.deepcopy(e.data.get_src_subset(e, state))
+                    src_subset.replace({self.loop.loop_variable: end})
+                    # None of write_subsets should lie within the new subset
+                    if any(intersects(ws_ss, src_subset) for ws_ss in write_subsets[dn.data]):
+                        return False
+
+        # No conditional edge may depend on the loop variable.
+        for edge in self.loop.all_interstate_edges():
+            if itervar_dep_syms.intersection(edge.data.condition.get_free_symbols()):
+                return False
+
+        # No conditional block may depend on the loop variable.
+        for node in self.loop.nodes():
```

Comment:
This should be checked recursively - it may be nested further down. Also what about loops depending on the loop variable?